### DISCUSSION TITLE
When the React project is redeployed, the new static files will overwrite the old static files

### DISCUSSION THREAD

**Discussion by 0x00Ethan** [vercel/community#5295] [Help] [answered]
### Summary

When the React project is redeployed, the new static files will overwrite the old static files, causing users to fail to access the old static files. How to deal with this problem?

### Steps to Reproduce

![260](https://github.com/vercel/community/assets/133326328/cd5ff3bd-b96e-4f22-a911-4afa49bf995f)

**Comment by leerob** [comment, marked as answer]
This is something that frameworks / bundlers can take care of for you. They can generate file names that contain unique hashes. Then, the unique file can be [safely cached](https://nextjs.org/docs/app/building-your-application/deploying#automatic-caching) [indefinitely on Vercel](https://vercel.com/docs/edge-network/caching#static-files-caching), and you can ensure that it doesn't break users visiting older versions of your application. This is how Next.js works by default, for example. 

![CleanShot 2023-12-26 at 11 13 12@2x](https://github.com/vercel/community/assets/9113740/4ecfa3e0-2cc5-412e-b005-872847e9069f)
